neo/Core/Blockchain.cs

```diff
@@ -341,24 +344,95 @@ public ECPoint[] GetValidators()
 
         public virtual IEnumerable<ECPoint> GetValidators(IEnumerable<Transaction> others)
         {
-            //TODO: 此处排序可能将耗费大量内存，考虑是否采用其它机制
-            VoteState[] votes = GetVotes(others).OrderBy(p => p.PublicKeys.Length).ToArray();
-            int validators_count = (int)votes.WeightedFilter(0.25, 0.75, p => p.Count.GetData(), (p, w) => new
-            {
-                ValidatorsCount = p.PublicKeys.Length,
-                Weight = w
-            }).WeightedAverage(p => p.ValidatorsCount, p => p.Weight);
-            validators_count = Math.Max(validators_count, StandbyValidators.Length);
-            Dictionary<ECPoint, Fixed8> validators = GetEnrollments().ToDictionary(p => p.PublicKey, p => Fixed8.Zero);
-            foreach (var vote in votes)
+            DataCache<UInt160, AccountState> accounts = GetStates<UInt160, AccountState>();
+            DataCache<ECPoint, ValidatorState> validators = GetStates<ECPoint, ValidatorState>();
+            MetaDataCache<ValidatorsCountState> validators_count = GetMetaData<ValidatorsCountState>();
+            foreach (Transaction tx in others)
             {
-                foreach (ECPoint pubkey in vote.PublicKeys.Take(validators_count))
+                foreach (TransactionOutput output in tx.Outputs)
+                {
+                    AccountState account = accounts.GetAndChange(output.ScriptHash, () => new AccountState(output.ScriptHash));
+                    if (account.Balances.ContainsKey(output.AssetId))
+                        account.Balances[output.AssetId] += output.Value;
```

Comment:
Maybe is faster use `TryGetValue`
With this way, you call internally get method with `ContainsKey` then call the get method for `+=` and then the set method

`IF([call GET]!=0) { [set][call GET]+[value] }`

With TryGetValue only one get is called because don't need the `+=`
`if[call GET to x]!=0) { [set][x]+[value] }`